cmd/janus/main.go

```diff
@@ -4,82 +4,72 @@ import (
 	"fmt"
 	"net/http"
 
-	"github.com/NYTimes/gziphandler"
+	mgo "gopkg.in/mgo.v2"
+
+	"net/url"
+
 	log "github.com/Sirupsen/logrus"
 	"github.com/hellofresh/janus/pkg/api"
-	"github.com/hellofresh/janus/pkg/jwt"
+	"github.com/hellofresh/janus/pkg/errors"
 	"github.com/hellofresh/janus/pkg/middleware"
 	"github.com/hellofresh/janus/pkg/oauth"
 	"github.com/hellofresh/janus/pkg/proxy"
 	"github.com/hellofresh/janus/pkg/router"
 	"github.com/hellofresh/janus/pkg/stats"
+	"github.com/hellofresh/janus/pkg/web"
 )
 
-//loadAPIEndpoints register api endpoints
-func loadAPIEndpoints(router router.Router, authMiddleware *jwt.Middleware) {
-	log.Debug("Loading API Endpoints")
-
-	// Apis endpoints
-	handler := api.NewController()
-	group := router.Group("/apis")
-	group.Use(authMiddleware.Handler, gziphandler.GzipHandler)
-	{
-		group.GET("", handler.Get())
-		group.POST("", handler.Post())
-		group.GET("/:id", handler.GetBy())
-		group.PUT("/:id", handler.PutBy())
-		group.DELETE("/:id", handler.DeleteBy())
-	}
-}
-
-//loadOAuthEndpoints register api endpoints
-func loadOAuthEndpoints(router router.Router, authMiddleware *jwt.Middleware) {
-	log.Debug("Loading OAuth Endpoints")
-
-	// Oauth servers endpoints
-	oAuthHandler := oauth.NewController()
-	oauthGroup := router.Group("/oauth/servers")
-	oauthGroup.Use(authMiddleware.Handler, gziphandler.GzipHandler)
-	{
-		oauthGroup.GET("", oAuthHandler.Get())
-		oauthGroup.POST("", oAuthHandler.Post())
-		oauthGroup.GET("/:id", oAuthHandler.GetBy())
-		oauthGroup.PUT("/:id", oAuthHandler.PutBy())
-		oauthGroup.DELETE("/:id", oAuthHandler.DeleteBy())
-	}
-}
-
-func loadAuthEndpoints(router router.Router, authMiddleware *jwt.Middleware) {
-	log.Debug("Loading Auth Endpoints")
-
-	handlers := jwt.Handler{Config: authMiddleware.Config}
-	router.POST("/login", handlers.Login())
-	authGroup := router.Group("/auth")
-	authGroup.Use(authMiddleware.Handler)
-	{
-		authGroup.GET("/refresh_token", handlers.Refresh())
-	}
-}
-
 func main() {
-	defer accessor.Close()
+	var repo api.APISpecRepository
+	var oAuthServersRepo oauth.Repository
+	var readOnlyAPI bool
+	var err error
+
 	defer statsdClient.Close()
 
 	statsClient := stats.NewStatsClient(statsdClient)
-
-	// create router
-	r := router.NewHttpTreeMuxRouter()
-	r.Use(
-		middleware.NewStats(statsClient).Handler,
-		middleware.NewLogger(globalConfig.Debug).Handler,
-		middleware.NewRecovery(RecoveryHandler).Handler,
-		middleware.NewMongoDB(accessor).Handler,
-	)
-
-	// create the proxy
-	oAuthServersRepo, err := oauth.NewMongoRepository(accessor.Session.DB(""))
-	if err != nil {
-		log.Panic(err)
+	dsnURL, err := url.Parse(globalConfig.Database.DSN)
+
+	switch dsnURL.Scheme {
+	case "mongodb":
+		log.Debugf("Trying to connect to %s", globalConfig.Database.DSN)
+		session, err := mgo.Dial(globalConfig.Database.DSN)
+		defer session.Close()
```

Comment:
session can be `nil` in some error cases, so defer must be after err check if err is `nil`